common/lib/xmodule/xmodule/modulestore/tests/test_publish.py

```diff
@@ -139,3 +157,1222 @@ def test_publish_draft_delete(self):
+        block_tree = ET.fromstring(block_contents)
+        # Now go through and verify the individual tags/attributes/children.
+        _assert_xml_level(block_tree, checklist['tag'], checklist.get('attrs'), checklist.get('children'))
+
+    def assertOLXContent(self, block_type, block_id, **kwargs):
+        """
+        Assert that a particular block exists in a course export in the proper draft/published location
+        and that the format is correct.
+        """
+        course_export_dir = self._get_course_export_dir()
+        is_draft = kwargs.pop('draft', False)
+        xml_to_check = kwargs.pop('xml', None)
+        xml_re_to_check = kwargs.pop('xml_re', None)
+        xml_parse = kwargs.pop('xml_parse', None)
+
+        block_path = self._get_block_type_path(course_export_dir, block_type, is_draft)
+        block_contents = self._get_block_contents(block_path, block_id)
+        if xml_to_check:
+            self.assertIn(xml_to_check, block_contents)
+        if xml_re_to_check:
+            xml_re = re.compile(xml_re_to_check)
+            self.assertIsNotNone(
+                xml_re.search(block_contents),
+                msg='Block ({}, {}) contents of:\n{}\n don\'t match regex of:\n{}'.format(
+                    block_type, block_id, block_contents, xml_re_to_check
+                )
+            )
+        if xml_parse:
+            self._assert_parsed_xml(block_contents, xml_parse)
+
+    def assertOLXMissing(self, block_type, block_id, **kwargs):
+        """
+        Assert that a particular block does not exist in a particular draft/published location.
+        """
+        course_export_dir = self._get_course_export_dir()
+        is_draft = kwargs.pop('draft', False)
+        block_path = self._get_block_type_path(course_export_dir, block_type, is_draft)
+        block_file_path = os.path.join(block_path, self._get_block_filename(block_id))
+        self.assertFalse(
+            os.path.exists(block_file_path),
+            msg='{} exists but should not!'.format(block_file_path)
+        )
+
+    def _make_deprecated_block_key(self, course_key, block_type, block_id):
+        """
+        Return a block key in the deprecated format.
+        """
+        return r'i4x://{ORG}/{COURSE}/{BLOCK_TYPE}/{BLOCK_ID}'.format(
+            ORG=course_key.org,
+            COURSE=course_key.course,
+            BLOCK_TYPE=block_type,
+            BLOCK_ID=block_id,
+        )
+
+    def _make_block_key(self, course_key, block_type, block_id):
+        """
+        Return a block key in the latest format.
+        """
+        return r'block-v1:{ORG}\+{COURSE}\+{RUN}\+type@{BLOCK_TYPE}\+block@{BLOCK_ID}'.format(
+            ORG=course_key.org,
+            COURSE=course_key.course,
+            RUN=course_key.run,
+            BLOCK_TYPE=block_type,
+            BLOCK_ID=block_id,
+        )
+
+    def _make_xml_parse_regex(self, block_type, course_key, draft, **kwargs):
+        """
```

Comment:
Hrm. I think my comment about having narrower functions applies, then. Rather than a dictionary with all of the tests to apply, it seems like individual assertions would be clearer.